### PR TITLE
:sparkles: Adding support to feature/Options

### PR DESCRIPTION
**Pull request by iturcios**
:sparkles: Adding support to feature/Options

### CODE REVIEW

File: src/main/java/faker/generator/FakeOptions.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Inherited
+@FakeServiceName(FakeOptionsService.class)
+public @interface FakeOptions {
+    FakeOptionsType value() default FakeOptionsType.YEAR;
+    int minDay() default 1;
+    int maxDay() default 31;
+    FakeOptionsMonths months() default FakeOptionsMonths.JANUARY;
+    int minYear() default 1900;
+    int maxYear() default 2020;
+    String[] days() default {"MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY"};
```
Comment (mauriciojovel) [inline src/main/java/faker/generator/FakeOptions.java]: This property should be required, please remove the default options. also rename the variable to value

File: src/main/java/faker/generator/FakeOptions.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+@Target(ElementType.FIELD)
+@Inherited
+@FakeServiceName(FakeOptionsService.class)
+public @interface FakeOptions {
+    FakeOptionsType value() default FakeOptionsType.YEAR;
+    int minDay() default 1;
+    int maxDay() default 31;
+    FakeOptionsMonths months() default FakeOptionsMonths.JANUARY;
+    int minYear() default 1900;
+    int maxYear() default 2020;
```
Comment (mauriciojovel) [inline src/main/java/faker/generator/FakeOptions.java]: Remove all this options, you need only the options

File: src/main/java/faker/generator/FakeOptionsMonths.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package faker.generator;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@Getter
+@AllArgsConstructor
+public enum FakeOptionsMonths {
+    JANUARY, FEBRUARY, MARCH, APRIL, MAY, JUNE, JULY, AUGUST, SEPTEMBER, OCTOBER, NOVEMBER, DECEMBER
+}
```
Comment (mauriciojovel) [inline src/main/java/faker/generator/FakeOptionsMonths.java]: Remove this file

File: src/main/java/faker/generator/FakeOptionsType.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package faker.generator;
+
+public enum FakeOptionsType {
+    DAY, MONTH, YEAR, DAY_OF_WEEK
+}
```
Comment (mauriciojovel) [inline src/main/java/faker/generator/FakeOptionsType.java]: Remove this file

File: src/main/java/faker/generator/services/FakeOptionsService.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                target.set(data, (faker.options().option(annotation.months().getDeclaringClass())).toString());
+                break;
+            case YEAR:
+                target.set(data, (faker.options().nextElement(IntStream.rangeClosed(annotation.minYear(),annotation.maxYear()).boxed().toArray())).toString());
+                break;
+            case DAY_OF_WEEK:
+                target.set(data, faker.options().nextElement(Arrays.asList(annotation.days())));
+                break;
+        }
+    }
```
Comment (mauriciojovel) [inline src/main/java/faker/generator/services/FakeOptionsService.java]: Rewrite the method to only use the new `value` property

File: src/test/java/faker/generator/models/Options.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    private String day;
+
+    @FakeOptions(FakeOptionsType.MONTH)
+    private String month;
+
+    @FakeOptions(FakeOptionsType.YEAR)
+    private String year;
+
+    @FakeOptions(FakeOptionsType.DAY_OF_WEEK)
+    private String dayOfWeek;
```
Comment (mauriciojovel) [inline src/test/java/faker/generator/models/Options.java]: Rewrite the model

File: src/main/java/faker/generator/FakeOptions.java (hunk 1 above)
Comment (iturcios) [reply src/main/java/faker/generator/FakeOptions.java]: Done.

File: src/main/java/faker/generator/FakeOptionsMonths.java (hunk 3 above)
Comment (iturcios) [reply src/main/java/faker/generator/FakeOptionsMonths.java]: Done.

File: src/main/java/faker/generator/FakeOptionsType.java (hunk 4 above)
Comment (iturcios) [reply src/main/java/faker/generator/FakeOptionsType.java]: Done.

File: src/main/java/faker/generator/services/FakeOptionsService.java (hunk 5 above)
Comment (iturcios) [reply src/main/java/faker/generator/services/FakeOptionsService.java]: Done.

File: src/test/java/faker/generator/models/Options.java (hunk 6 above)
Comment (iturcios) [reply src/test/java/faker/generator/models/Options.java]: Done.

File: src/main/java/faker/generator/FakeOptions.java (hunk 2 above)
Comment (iturcios) [reply src/main/java/faker/generator/FakeOptions.java]: Done.

File: src/main/java/faker/generator/FakeOptions.java [hunk 7]
```diff
@@ -11,11 +11,5 @@
...
 @FakeServiceName(FakeOptionsService.class)
 public @interface FakeOptions {
-    FakeOptionsType value() default FakeOptionsType.YEAR;
-    int minDay() default 1;
-    int maxDay() default 31;
-    FakeOptionsMonths months() default FakeOptionsMonths.JANUARY;
-    int minYear() default 1900;
-    int maxYear() default 2020;
-    String[] days() default {"MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY"};
+    String[] value();
```
Comment (mauriciojovel) [inline src/main/java/faker/generator/FakeOptions.java]: Please change for objects, to be more generic

Comment (iturcios) [reply src/main/java/faker/generator/FakeOptions.java]: As we talked, I changed for String, int and double arrays.

**Pull request merged**